sycl/doc/developer/WorkingOnAReleaseBranch.md

```diff
@@ -0,0 +1,39 @@
+# Working on a release branch
+
+Release branch is defined as a branch whose name starts with `sycl-rel-` prefix.
+
+Those branches are intended to indicate stable snapshots of our product so that
+our users don't need to guess which nightly build is good enough for our needs.
+
+Therefore, those branches have higher quality requirements and as such have
+different contribution rules intended to preserve their stability.
+
+If you are not familiar with [general contribution guidelines][contributing] or
+[DPC++ specific contribution guidelines][contributing-to-dpcpp], please
+familiarize yourself with those documents first because they also apply to
+release branches.
+
+## Extra rules for release branches
+
+### Only cherry-picks are allowed
+
+It is assumed, that everything you do on a release branch, should also be
+repeated on the main branch to ensure that it is automatically included into
```

Comment:
Perhaps this should be *the default branch* instead of *the main branch* to avoid confusion with the upstream main branch which exists but is not the default?

Reply:
Addressed this in d3a51ea82ac6b2c08c22eed17e4b5ddfec41f1f8